crates/rustc_codegen_spirv/src/builder/builder_methods.rs

```diff
@@ -315,7 +338,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         } else {
             for index in 0..count {
                 let const_index = self.constant_u32(self.span(), index as u32);
-                let gep_ptr = self.gep(pat.ty, ptr, &[const_index]);
+                let gep_ptr = self.inbounds_gep(pat.ty, ptr, &[const_index]);
```

Comment:
I added this because I assume that const memsets will always be in a valid range, but I'm not sure.